core/src/main/kotlin/com/google/container/tools/core/analytics/UsageTrackerManagerService.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.container.tools.core.analytics
+
+import com.google.common.annotations.VisibleForTesting
+import com.google.container.tools.core.properties.PropertiesFileFlagReader
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.components.ServiceManager
+
+/**
+ * Application service that manages the status of usage tracking in the plugin.
+ */
+class UsageTrackerManagerService(
+    private val propertyReader: PropertiesFileFlagReader = PropertiesFileFlagReader()
```

Comment:
This looks a bit heavy for a single value dependency - could we use `analyticsId` as a parameter directly instead?

Reply:
And, conceivably, just expose `PropertiesFileFlagReader` as a simple service?

Reply:
Could you clarify what part of this is heavy? Do you mean the UsageTrackerManagerService class in general?

Reply:
I mean we pass a whole property file reader to this object just to extract a single string value, and it seems we don't anticipate many new properties or access to a reader itself in this class which is why it's a bit confusing to see we pass a reader.

How about having plugin properties as a service or just a singleton companion object, and then just pass in actual analytics id value as a parameter?

Reply:
Ah. Yeah I have it passed in because I wanted to mock it for testing. Changing it to a service would work.

Wrt passing the analytics id value in. I think that will be fine. I'll try and refactor.